okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -356,10 +351,20 @@ public void run() {
       try {
         // Read until the underlying socket closes.
         while (frameReader.nextFrame(this)) {
+          // No need synchronization here, since frameReader is not closed
```

Comment:
Synchronization is necessary to see the updated value. It _can sometimes_ work without the synchronization, but the only way to guarantee that this thread ever sees an update to stopped is via synchronization/volatile.